packages/direflow-component/src/services/asyncScriptLoader.ts

```diff
@@ -0,0 +1,50 @@
+type TBundle = { script: Element; hasLoaded: boolean };
+declare global {
+  interface Window {
+    wcPolyfillsLoaded: TBundle[];
+    reactBundleLoaded: TBundle[];
+  }
+}
+
+const asyncScriptLoader = (src: string, bundleList: TBundle[] | undefined): Promise<void> => {
```

Comment:
why not make this `bundleList: TBundle[] | undefined`

Reply:
I think this is about communication.
`bundleList?:` would imply that the argument is optional.
That is not the case - but it’s gonna be a reference to a list on the global object that may not have been initialized yet, hence carries the _value_ of undefined.

If you look down a little further, you will see that it's getting initialized if not already set:
```javascript
if (!bundleList) {
  bundleList = [];
}
```